docs/theme/mlflow/static/js/version-dropdown.js

```diff
@@ -0,0 +1,55 @@
+fetch('https://pypi.org/pypi/mlflow/json')
+  .then((response) => response.json())
+  .then((data) => {
+    var versions = Object.keys(data.releases)
+      // Drop dev/post/rc versions
+      .filter(function (version) {
+        return /^\d+(\.\d+){0,3}$/.test(version);
+      })
+      // Sort versions
+      // https://stackoverflow.com/a/40201629
+      .map((a) =>
+        a
+          .split('.')
+          .map((n) => +n + 100000)
+          .join('.'),
+      )
+      .sort()
+      .map((a) =>
+        a
+          .split('.')
+          .map((n) => +n - 100000)
+          .join('.'),
+      )
+      .reverse();
+
+    var latestMicroVersions = [];
+    versions.forEach(function (version) {
+      var microVersion = version.split('.').slice(0, 2).join('.');
+      if (!latestMicroVersions.includes(microVersion)) {
+        latestMicroVersions.push(version);
+      }
+    });
+
+    var latestVersion = latestMicroVersions[0];
+    var docRegex = /\/docs\/(?<version>[^/]+)\//;
+    var currentVersion = docRegex.exec(window.location.pathname).groups.version;
+    var dropDown = document.createElement('select');
+    dropDown.onchange = function () {
+      var newHref = window.location.href.replace(docRegex, `/docs/${this.value}/`);
+      window.location.href = newHref;
```

Comment:
```suggestion
      window.location.assign(docRegex, `/docs/${this.value}/`);
```
Would this enable using the back button / history? Doesn't a `.replace()` overwrite the current page in browser history?
I checked a few other python docs sites that have this feature of selecting versioning and they all have navigation persisted in the browser history.

Reply:
That comparison chart is so much nicer than searching through the react docs :)